enhancements/sig-policy/85-gatekeeper-policy-integration/README.md

```diff
@@ -0,0 +1,257 @@
+      name: gatekeeper-system
+```
+
+The status message for the constraint in the OCM `Policy` would be in the format of
+`<message> (<enforcementAction> of <kind> <namespace/name or name>)` for each violation entry each separated by a
+semicolon. For example, the above example would produce a status message of
+`you must provide labels: {"gatekeeper"} (dryrun of Namespace default); you must provide labels: {"gatekeeper"} (dryrun of Namespace gatekeeper-system)`.
+
+If Gatekeeper is not installed, the OCM `Policy` will be marked as non-compliant with a message stating that the user
+needs to install Gatekeeper to apply such a constraint.
+
+#### Policy Deletion
+
+When an OCM `Policy` embeds a Gatekeeper `ConstraintTemplate`/constraint is deleted, the embeded Gatekeeper
+`ConstraintTemplate`/constraint will be deleted to match the behavior of other OCM policy templates.
+
+### User Stories
+
+#### Story 1
+
+As a policy user, I would like to have native multicluster support for an admission controller so that I can have a
+consistent distribution and status aggregation mechanism for all my policies.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Template Sync Updates
+
+The Template Sync controller is a controller in the `governance-policy-framework-addon` controller manager responsible
+for creating and updating policies on the managed cluster that are defined in the `Policy` object's `policy-templates`
+array. So far, all these policies are namespace scoped. Gatekeeper `ConstraintTemplate` objects and constraint objects
+are cluster scoped.
+
+The changes required to support cluster scoped policy templates would mean that the Template Sync controller will need
+to detect when a policy template is namespaced before querying or creating it. Additionally, owner references are
+currently relied upon to properly delete the policy templates when the parent policy is deleted. Since namespaced
+objects can't own cluster scoped objects in Kubernetes, an alternative is required.
+
+The Template Sync controller would not put owner references on Gatekeeper `ConstraintTemplate`/constraints, but instead
+add a finalizer of `policy.open-cluster-management.io/cleanup-cluster-scoped-policies` to the `Policy` object that
+defines the Gatekeeper `ConstraintTemplate`/constraints. This finalizer would indicate that Gatekeeper
+`ConstraintTemplate`/constraints need to be removed when the OCM `Policy` object is deleted and if applicable, when the
+`Policy` object is updated. To be able to track which Gatekeeper constraints belong to a `Policy` object, a new label
+should be added to the Gatekeeper `ConstraintTemplate` objects and constraints of
+`policy.open-cluster-management.io/policy` with the policy name. Note that a finalizer will also need to be added to the
+`governance-policy-framework` `Deployment` like the Configuration Policy controller does to prevent the Template Sync
+from being deleted before the finalizer can be handled.
```

Comment:
I think we should make sure this works similarly to our fix for https://issues.redhat.com/browse/ACM-3049 (which I think @willkutler is thinking about).